vital/types/local_cartesian.h

```diff
@@ -0,0 +1,182 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief This file contains the interface to a local geographic offset coordinate system.
+ */
+
+#ifndef KWIVER_VITAL_LOCAL_CARTESIAN_H_
+#define KWIVER_VITAL_LOCAL_CARTESIAN_H_
+
+#include <vital/types/point.h>
+#include <vital/types/geo_point.h>
+
+namespace kwiver {
+namespace vital {
+
+// ----------------------------------------------------------------------------
+/** Local Cartesian Conversion Utility.
+ *
+ * Base on the NGA GeoTrans library
+ * https://earth-info.nga.mil/GandG/update/index.php?action=home
+ *
+ * This class is a cleaned up version of the LocalCartesian and Geocentric classes provided in GeoTrans
+ * This allows a user to define a local cartesian coordinate system with any origin (expressed in WGS84)
+ */
+class VITAL_EXPORT local_cartesian
+{
+public:
+
+  local_cartesian();
+  virtual ~local_cartesian() = default;
+
+  /**
+   * Set local origin parameters as inputs and sets the corresponding state variables.
+   * NOTE : If the origin changes, this method needs to be called again to recompute
+   *        variables needed in the conversion math.
+   *
+   *    geo_point                : Geograpical origin of the cartesian system        (input)
+   *    orientation              : Orientation angle of the local cartesian coordinate system,
+   *                               in radians                                        (input)
+   */
+  void set_origin(geo_point const& origin, double orientation=0);
```

Comment:
It seems like you would want a constructor that directly takes the origin point.  You may even want to disable the default constructor. What use case is there for a default constructed and invalid coordinate system?  Is there even a way to check that it is not initialized?

Reply:
I will add a way to ensure the origin has been set and the data is valid.
And throw an error if you try to convert with out setting an origin

A ctor with the same spec as set_origin is a good idea

I like having a default ctor (assuming I can ensure this class has a valid origin) so you can make an instance of the converter and reuse it to process point on multiple origins if you need to

Reply:
I don't see how a default ctor helps with that.  I don't see this as a class where you would be reusing the same instance and changing the origin point to operate on other points.  I think you would just have multiple instances of this class, one for each origin for each group of associated points.

Reply:
I would think something like a method that processes a large file of already created tracks provided in world coordinates (like what we got on BIGT4) and generating track states and setting the tracking points by computing the cartesian of provided coordinate and a given origin

Which probably would not require moving the origin around... but it seems like its always a good idea to have a default ctor on classes... just in case...

I made a 'valid_origin' method to ensure there is an origin set

Doesn't hurt, and might be useful in the future... 
or maybe we are just providing a way for someone to shoot themselves in the foot

Reply:
I think we are making a way for someone to shoot themselves in the foot by allowing an invalid coordinate system.

Reply:
All right, i will nix the default ctor (easy to add if somebody ever needs it for whatever reason)